common/hashing.h

```diff
@@ -16,28 +16,58 @@
 #include "llvm/ADT/SmallVector.h"
 #include "llvm/ADT/StringRef.h"
 #include "llvm/Support/FormatVariadic.h"
+#include "llvm/Support/MathExtras.h"
+
+#ifdef __ARM_ACLE
+#include <arm_acle.h>
+#endif
 
 namespace Carbon {
 
-// A type wrapping a 64-bit hash code and provides a basic but limited API.
+// A 64-bit hash code produced by `Carbon::HashValue`.
+//
+// This provides methods for extracting bits from the hash code that work to be
+// fast but extract the highest quality bits.
```

Comment:
Sorry about breaking this up. Multi-line comments are broken for me on GitHub.

```suggestion
// quickly.
```

Reply:
Done.